history/taskana-simplehistory-provider/src/test/java/acceptance/events/task/CreateHistoryEventOnTaskCancelClaimAccTest.java

```diff
@@ -48,8 +53,44 @@ void should_CreateCancelClaimedHistoryEvent_When_TaskIsCancelClaimed() throws Ex
 
     assertThat(events).hasSize(1);
 
-    String eventType = events.get(0).getEventType();
+    TaskHistoryEvent event = events.get(0);
+    String eventType = event.getEventType();
 
     assertThat(eventType).isEqualTo(TaskHistoryEventType.CLAIM_CANCELLED.getName());
+
+    event = historyService.getTaskHistoryEvent(event.getId());
+
+    assertThat(event.getDetails()).isNotNull();
+
+    JSONArray changes = new JSONObject(event.getDetails()).getJSONArray("changes");
+    System.out.println(changes.toString());
+    JSONObject expectedClaimed =
+        new JSONObject()
+            .put("newValue", "")
+            .put("fieldName", "claimed")
+            .put("oldValue", oldClaimed.toString());
+    JSONObject expectedModified =
+        new JSONObject()
+            .put("newValue", task.getModified().toString())
+            .put("fieldName", "modified")
+            .put("oldValue", oldModified.toString());
+    JSONObject expectedState =
+        new JSONObject()
+            .put("newValue", "READY")
+            .put("fieldName", "state")
+            .put("oldValue", "CLAIMED");
+    JSONObject expectedOwner =
+        new JSONObject().put("newValue", "").put("fieldName", "owner").put("oldValue", "user-b-1");
+
+    JSONArray expectedChanges =
+        new JSONArray()
+            .put(expectedClaimed)
+            .put(expectedModified)
+            .put(expectedState)
+            .put(expectedOwner);
```

Comment:
personally I don't like this way of creating the JSONArray instance.. But since Java is full of boilerplate we have no other choice for now .. Everything else would be too much effort.

This is basically just for you: It's way too verbose for comparing the two JSON Objects